src/components/shared/InnerNavigationLink.js

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+
+import { setDataSource } from 'firefox-profiler/actions/profile-view';
+
+import explicitConnect, {
+  type ConnectedProps,
+} from 'firefox-profiler/utils/connect';
+
+import type { DataSource } from 'firefox-profiler/types';
+
+type OwnProps = {|
+  +className?: string,
+  +dataSource: DataSource,
+  +children: React.Node,
+|};
+
+type DispatchProps = {|
+  +setDataSource: typeof setDataSource,
+|};
+
+type Props = ConnectedProps<OwnProps, {||}, DispatchProps>;
+
+class InnerNavigationLinkImpl extends React.PureComponent<Props> {
+  onClick = (e: SyntheticMouseEvent<>) => {
+    const { setDataSource, dataSource } = this.props;
+    if (e.ctrlKey || e.metaKey) {
+      // The user clearly wanted to open this link in a new tab.
+      return;
+    }
+
+    e.preventDefault();
+
+    setDataSource(dataSource);
```

Comment:
No action needed in this PR:

We really need to rename this `DataSource`, as it's being used for things that aren't sources for a profile, but rather the current page. I got very confused by it in another patch I was working on. I think maybe `PageSlug` might work.

Reply:
Agreed!